src/AWS.Deploy.CLI/Commands/ServerModeCommand.cs

```diff
@@ -91,5 +120,29 @@ private IEncryptionProvider CreateEncryptionProvider()
 
             return encryptionProvider;
         }
+
+        public class MonitoringExited
+        {
+            public bool ProcessNotFound {get;set;}
+        }
+
+        private async Task<MonitoringExited> WaitOnParentPid(CancellationToken token)
+        {
+            if (_parentPid == null)
+                return new MonitoringExited { ProcessNotFound = true };
+
+            while (true)
+            {
+                try
+                {
+                    Process.GetProcessById((int)_parentPid);
+                    await Task.Delay(1000, token);
+                }
+                catch
+                {
+                    return new MonitoringExited { ProcessNotFound = true };
```

Comment:
Is there a specific exception that you can catch?

Reply:
There is, but the method throws multiple exceptions. I want to consider any issue retrieving the parent process as the same issue for our context. If anything goes wrong, then the parent process is not available.

Reply:
Then I think we don't need

```
        public class MonitoringExited
        {
            public bool ProcessNotFound {get;set;}
        }
```
Setting ProcessNotFound to true gives a false indication. We should simply return a boolean that indicates the whether GetProcessById threw an exception or not.

If needed we can always expand return type but I don't reason for MonitoringExited existence.

Reply:
I removed MonitoringExited and returned a bool.